EIPS/eip-4444.md

```diff
@@ -0,0 +1,101 @@
+---
+eip: 4444
+title: Bound Historical Data in Execution Clients
+description: Prune historical data in clients older than one year
+author: George Kadianakis (@asn_d6), lightclient (@lightclient), Alex Stokes (@ralexstokes)
+discussions-to: TBD
+status: Draft
+type: Standards Track
+category: Core
+created: 2021-11-02
+---
+
+## Abstract
+
+Clients must stop serving historical blocks/receipts that are older than a year on the p2p layer. Clients may prune such historical data.
+
+## Motivation
+
+Historical blocks, states, and receipts currently occupy more than 400GB of disk space (and growing!). Therefore, to validate the chain, users must typically have a 1TB disk.
+
+Historical data is not necessary for validating new blocks, so once a client has synced the tip of the chain, historical data is only retrieved when requested explicitly over the JSON-RPC or when a peer attempts to sync the chain. By pruning the history, this proposal reduces the disk requirements for users. Pruning history also allows clients to remove code that processes historical blocks. This means that execution clients don't need to maintain code paths that deal with each upgrade's compound changes.
+
+This proposal also concretizes the availability guarantees of the p2p layer as clients won't be able to fetch historical data anymore. This results in less strain on the network as clients adopt more lightweight sync strategies based on the PoS weak subjectivity assumption.
+
+## Specification
+
+| Parameter | Value | Description |
+| - | - | - |
+| `HISTORY_PRUNE_EPOCHS` | 82125 | A year in beacon chain epochs |
+
+Clients MUST NOT serve blocks/states/receipts that are older than `HISTORY_PRUNE_EPOCHS` epochs on the p2p network.
```

Comment:
Not everyone will agree on what time it is, should we have a bit of a no-mans-land between the time clients are expected to stop *requesting* ancient data and when clients are expected to stop *providing* ancient data?  This could be minutes/hours/days, just enough to deal with time synchronization issues.

Presumably, under such a regime any peer that requests data older than the longer of the two **SHOULD** be kicked.  Meanwhile, clients **SHOULD** be programmed to never ask for anything older than the shorter of the two.

Reply:
So I can/should locally prune after these epochs, and thus I can be in a situation where our clocks have a disparity  (within reason) and you make what you think is an honest request that I do not have.  Are you suggesting in such a situation that although I cannot respond to the request, I *should not* consider your request as nefarious? Whereas if you are outside of the window, I can consider this as "bad" behavior and descore/kick you?

> under such a regime any peer that requests data older than the longer of the two SHOULD be kicked

I prefer **MAY** be downscored and/or kicked.

Reply:
> Not everyone will agree on what time it is, should we have a bit of a no-mans-land between the time clients are expected to stop _requesting_ ancient data and when clients are expected to stop _providing_ ancient data? This could be minutes/hours/days, just enough to deal with time synchronization issues.
> 
> Presumably, under such a regime any peer that requests data older than the longer of the two **SHOULD** be kicked. Meanwhile, clients **SHOULD** be programmed to never ask for anything older than the shorter of the two.

@MicahZoltu In my country, MasterCard and Visa while not blockchain related are required to keep past history of the last 10 years for law enforcement and they use a higher transaction throughout.